#!/usr/bin/env python
PACKAGE = "turtle_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("KP_param", double_t, 0, "KP parameter",    0, 0,   10)
gen.add("KI_param", double_t, 0, "KI parameter",    0, 0,   400)
gen.add("KD_param", double_t, 0, "KD parameter",    0, 0,   10)
gen.add("Sample_time", double_t, 0, "sample time parameter",    0.2, 0,   1)
gen.add("Speed",    int_t,    0, "Speed parameter", 0,  0, 676)

size_enum = gen.enum([ gen.const("Low",      int_t, 0, "Low is 0"),
                       gen.const("Medium",     int_t, 1, "Medium is 1"),
                       gen.const("High",      int_t, 2, "Hight is 2")],
                     "Select from the list")

gen.add("size", int_t, 0, "Select from the list", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "turtle_driver", "pid_control"))
